### PR TITLE
0. Toolchain Setup

### PR DESCRIPTION
**Pull request by bitschmidty**
Combined [James](https://github.com/bitcoinops/taproot-workshop/pull/5) and [Johns](https://github.com/bitcoinops/taproot-workshop/pull/4) PRs to start.

Work in progress, more to come.

- [x] Fix the check on bitcoind version. Currently fails.

### CODE REVIEW

**Comment by jachiang** [comment]
ACK, notebook runs without errors with [v0.1](https://github.com/bitcoinops/bitcoin/tree/v0.1).

File: 0.1-test-notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Source directory configured as /Users/schmidty/Dropbox/Repos/bitcoin-optech-taproot\n"
+     ]
+    }
+   ],
+   "source": [
+    "import setup"
```
Comment (jachiang) [inline 0.1-test-notebook.ipynb]: Update to `import util` when [TestWrapper PR](https://github.com/bitcoinops/taproot-workshop/pull/8) is resolved.

File: 0.1-test-notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2019-08-29T19:30:18.841000Z TestFramework./var/folders/fq/cbxk_q2n2sb1hw8zxl987v7h0000gn/T/bitcoin_func_test_yedz4z_d (INFO): Initializing test directory /var/folders/fq/cbxk_q2n2sb1hw8zxl987v7h0000gn/T/bitcoin_func_test_yedz4z_d\n"
+     ]
+    }
+   ],
+   "source": [
+    "from test_framework.test_wrapper import TestWrapper\n",
```
Comment (jachiang) [inline 0.1-test-notebook.ipynb]: If using util.py, this import can be removed.

File: 0.1-test-notebook.ipynb (hunk 1 above)
Comment (jnewbery) [reply 0.1-test-notebook.ipynb]: Done. I've rebased this PR on the TestWrapper changes.

File: 0.1-test-notebook.ipynb (hunk 2 above)
Comment (jnewbery) [reply 0.1-test-notebook.ipynb]: Done. I've rebased this PR on the TestWrapper changes.

File: README.md [hunk 3]
```diff
@@ -27,22 +25,69 @@ standard bitcoin data directory directory, but why risk it?
...
 ## Requirements
 
-#### bitcoind
+#### Build a taproot-supporting bitcoind
+
+These workbooks require a `bitcoind` built from the [Optech taproot
+branch, v0.1](https://github.com/bitcoinops/bitcoin/releases/tag/v0.1) which
+supports schnorr and taproot. 
+
+![workshop_repositories](files/0-repositories-diagram.jpg)
```
Comment (jnewbery) [inline README.md]: See comment: https://github.com/bitcoinops/taproot-workshop/pull/5#discussion_r318651982

- I think the 'imports' arrow is going the wrong way. Either change the text to 'imported by' or reverse the arrow.
- s/1.0-Workshop-Setup.iypnb/1.0-Workshop-Setup.ipynb/

File: 0.1-test-notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Things to keep in mind:*\n",
+    "* Separate test wrapper `setup`, `shutdown` and your code into different cells. This ensures that an error in your code does not affect the test wrapper object and prevents orphaned subprocesses.\n",
+    "* Only run one test wrapper object at a time.\n",
+    "* If the subprocesses do become orphaned somehow, you can run the following:\n",
+    " * `ps -ax | grep bitcoind` to see outstanding bitcoind processes.\n",
```
Comment (jnewbery) [inline 0.1-test-notebook.ipynb]: Do we also need instructions for windows users here?

Comment (jnewbery) [reply 0.1-test-notebook.ipynb]: Tracked here: https://github.com/bitcoinops/taproot-workshop/issues/14

**Comment by jnewbery** [comment]
I've rebased on master and pushed a commit with a bunch of minor nits.

I'm going to merge this now. Any further feedback can be addressed in follow-up PRs.

**Pull request merged**